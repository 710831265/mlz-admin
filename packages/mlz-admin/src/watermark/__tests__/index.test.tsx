import React, { Component } from 'react';
import { testMount, testSnapshot, sleep } from '../../../../../tests';
import Watermark, { errorThrower } from '..';
import { mount } from 'enzyme';

jest.mock('../canvas', () => {
  return {
    drawText: jest.fn(),
  };
});

describe('ðŸ§ª DarkThemeToggler', () => {
  const props = {
    text: 'æµ‹è¯•',
    children: <span>1234</span>,
  };
  testMount(Watermark, props);
  testSnapshot(Watermark, props);

  test('Watermarkç»„ä»¶çš„childrenå¿…é¡»å”¯ä¸€', async () => {
    let error = null;
    try {
      mount(
        // @ts-ignore
        <Watermark text="æµ‹è¯•">
          <span id="s1" />
          <span id="s2" />
        </Watermark>,
      );
    } catch (err) {
      error = err;
    }

    expect(error).toMatch(/Watermark should have only one children, but got 2/);
  });
});
