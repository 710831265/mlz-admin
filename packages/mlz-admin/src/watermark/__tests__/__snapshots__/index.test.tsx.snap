// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🧪 DarkThemeToggler snapshot test 快照匹配 1`] = `"<span style=\\"background: transparent url(data:image/png;base64,00);\\">1234</span><canvas style=\\"position: absolute; z-index: -999; display: none;\\" width=\\"200\\"></canvas>"`;
